cli/tests/test_workspaces.rs

```diff
@@ -712,6 +709,65 @@ fn test_workspaces_forget_multi_transaction() {
     "###);
 }
 
+#[test]
+fn test_workspaces_forget_abandon_commits() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_ok(test_env.env_root(), &["git", "init", "main"]);
+    let main_path = test_env.env_root().join("main");
+
+    std::fs::write(main_path.join("file"), "contents").unwrap();
+
+    test_env.jj_cmd_ok(&main_path, &["workspace", "add", "../second"]);
+    test_env.jj_cmd_ok(&main_path, &["workspace", "add", "../third"]);
+    test_env.jj_cmd_ok(&main_path, &["workspace", "add", "../fourth"]);
+    let third_path = test_env.env_root().join("third");
+    test_env.jj_cmd_ok(&third_path, &["edit", "second@"]);
+    let fourth_path = test_env.env_root().join("fourth");
+    test_env.jj_cmd_ok(&fourth_path, &["edit", "second@"]);
+
+    // there should be four workspaces, three of which are at the same empty commit
+    let stdout = test_env.jj_cmd_success(&main_path, &["workspace", "list"]);
+    insta::assert_snapshot!(stdout, @r###"
+    default: qpvuntsm 4e8f9d2b (no description set)
+    fourth: uuqppmxq 57d63245 (empty) (no description set)
+    second: uuqppmxq 57d63245 (empty) (no description set)
+    third: uuqppmxq 57d63245 (empty) (no description set)
+    "###);
+    insta::assert_snapshot!(get_log_output(&test_env, &main_path), @r###"
+    ◉  57d63245a308 fourth@ second@ third@
+    │ @  4e8f9d2be039 default@
+    ├─╯
+    ◉  000000000000
+    "###);
+
+    // delete the default workspace (should not abandon commit since not empty)
+    test_env.jj_cmd_ok(&main_path, &["workspace", "forget", "default"]);
```

Comment:
nit: can you also add snapshot of `stderr`?

Reply:
Since the `stderr` is empty, I just switched it to use `jj_cmd_success`.